Refresh.GameServer/Services/AipiService.cs

```diff
@@ -0,0 +1,161 @@
+        AipiResponse<TData>? aipiResponse = await response.Content.ReadFromJsonAsync<AipiResponse<TData>>();
+        
+        if (aipiResponse == null) throw new Exception("No response was received from the server.");
+        if (!aipiResponse.Success) throw new Exception($"{response.StatusCode}: {aipiResponse.Reason}");
+
+        return aipiResponse.Data!;
+    }
+    
+    private async Task<Dictionary<string, float>> PredictEvaAsync(Stream data)
+    {
+        Stopwatch stopwatch = new();
+        this.Logger.LogTrace(RefreshContext.Aipi, "Pre-processing image data...");
+
+        DecoderOptions options = new()
+        {
+            MaxFrames = 1,
+            Configuration = SixLabors.ImageSharp.Configuration.Default,
+        };
+
+        Image image = await Image.LoadAsync(options, data);
+        // Technically, we don't read videos in Refresh like in DO, but a couple of users are currently using APNGs as their avatar.
+        // I don't want to break APNGs as they're harmless, so let's handle this by just reading the first frame for now.
+        if (image.Frames.Count > 0)
+            image = image.Frames.CloneFrame(0);
+        
+        image.Mutate(x => x.Resize(new ResizeOptions
+        {
+            Size = new Size(512),
+            Mode = ResizeMode.Max,
+        }));
+
+        using MemoryStream processedData = new();
+        await image.SaveAsPngAsync(processedData);
+        // await image.SaveAsPngAsync($"/tmp/{DateTimeOffset.Now.ToUnixTimeMilliseconds()}.png");
+        processedData.Seek(0, SeekOrigin.Begin);
+
+        float threshold = this._config.AipiThreshold;
+        
+        this.Logger.LogDebug(RefreshContext.Aipi, $"Running prediction for image @ threshold={threshold}...");
+
+        stopwatch.Start();
+        Dictionary<string, float> prediction = await this.PostAsync<Dictionary<string, float>>($"/eva/predict?threshold={threshold}", processedData);    
+        stopwatch.Stop();
+
+        this.Logger.LogInfo(RefreshContext.Aipi, $"Got prediction result in {stopwatch.ElapsedMilliseconds}ms.");
+        this.Logger.LogDebug(RefreshContext.Aipi, JsonConvert.SerializeObject(prediction));
+        return prediction;
+    }
+
+    public bool ScanAndHandleAsset(DataContext context, GameAsset asset)
+    {
+        // guard the fact that assets have an owner
+        Debug.Assert(asset.OriginalUploader != null, $"Asset {asset.AssetHash} had no original uploader when trying to scan");
+        if (asset.OriginalUploader == null)
+            return false;
+
+        // import the asset as png
+        bool isPspAsset = asset.AssetHash.StartsWith("psp/");
+        string hash = isPspAsset ? "psp/" + asset.AssetHash : asset.AssetHash;
+
+        if (!context.DataStore.ExistsInStore("png/" + asset.AssetHash))
+        {
+            this._importer.ImportAsset(asset.AssetHash, isPspAsset, asset.AssetType, context.DataStore);
+        }
```

Comment:
That `hash` variable is unused, and I believe can be removed, given `ImportAsset` will handle PSP assets for us.

I'm also really starting to dislike that `psp` is something we have to prepend everywhere, it really feels like a hack. It's out of scope for this PR, but at some point I'm going to find a modular solution to the issue of "game locked assets".

Reply:
Didn't catch that.